src/Distributed/distributed_incompressible_model.jl

```diff
@@ -0,0 +1,97 @@
+using MPI
+
+using Oceananigans
+using Oceananigans.Grids
+using Oceananigans.Grids: halo_size
+
+#####
+##### Distributed model struct and constructor
+#####
+
+struct DistributedIncompressibleModel{A, G, M}
+    architecture :: A
+            grid :: G
+           model :: M
+end
+
+function DistributedIncompressibleModel(; architecture, grid, boundary_conditions=nothing, model_kwargs...)
+    my_rank = architecture.my_rank
+    i, j, k = architecture.my_index
+    Rx, Ry, Rz = architecture.ranks
+    my_connectivity = architecture.connectivity
+
+    ## Construct local grid
+
+    Nx, Ny, Nz = size(grid)
+
+    # Pull out left and right endpoints for full model.
```

Comment:
It's only left and right in `x`   Maybe say # Pull out endpoints for full model.

Reply:
True. Sometimes we use "left" and "right" for x, y, z in the code, and x gets "east" and "west", y gets "north" and "south" etc. but your suggestion is clearer so I've made the change.